docs/canary_releases.md

```diff
@@ -0,0 +1,70 @@
+# Canary releases
+
+There is a need to use canary releases in order to make sure newer versions
+of the gathering conditions don't break the Insights Operator gatherer.
+
+This is how a canary release would look like:
+
+```mermaid
+sequenceDiagram
+    actor RT as Rules Team
+    actor PT as Processing Team
+    participant GC as Gathering Conditions
+    participant AI as App Interface
+    participant GCS as Gathering Conditions Service
+    participant Unleash
+    participant Storage
+    actor C as Cluster
+    actor CC as Canary Cluster
+
+    Note left of GCS: Latest version is 1.0.1
+    GCS -->> C: Get version 1.0.1
+    GCS -->> CC: Get version 1.0.1
+
+    RT ->> GC: Push a new tag 1.0.2
+    RT ->> AI: Set the canary and stable versions<br/>CANARY_VERSION = 1.0.2<br/>STABLE_VERSION = 1.0.1
+    PT ->> AI: Review and merge the PR
+    AI ->> GCS: Update prod deployment
+    
+    rect rgb(250,128,114)
+        alt canary or stable version not found
+            GCS ->> GCS: CrashLoopBackOff
+            PT ->> AI: Revert the PR
+        end
+    end
+
+    GCS -->> C: Get version 1.0.1
+    GCS -->> CC: Still get version 1.0.1
```

Comment:
This assumes the canary population setting was at 0%, but the "canary archives are wrong" doesn't reset it.

Reply:
No, even if the canary population is at X%, being X !=0, as the PR was reverted, all clusters, be it canary or normal, will receive the stable version (1.0.1).

Reply:
That wasn't the point. The point was that if you roll out the canary without checking the Unleash configuration, the canary version would be served immediately to an unknown percentage of clusters (whatever percentage was left there from the previous deployment). We don't have to reset the percentage to 0, but checking what the percentage is should be a documented step in the process.